src/lib/utils/meta_tables/abstract_meta_system_table.cpp

```diff
@@ -0,0 +1,46 @@
+#ifdef __APPLE__
+#include <mach/mach.h>
+#endif
+
+#include <fstream>
+
+#include "utils/meta_tables/abstract_meta_system_table.hpp"
+
+namespace opossum {
+
+AbstractMetaSystemTable::AbstractMetaSystemTable(const TableColumnDefinitions& column_definitions)
+    : AbstractMetaTable(column_definitions) {}
+
+// Returns the number of logical processors
+size_t AbstractMetaSystemTable::_get_cpu_count() {
+#ifdef __linux__
+
+  std::ifstream cpu_info_file;
+  cpu_info_file.open("/proc/cpuinfo", std::ifstream::in);
+  DebugAssert(cpu_info_file.is_open(), "Failed to open /proc/cpuinfo");

```

Comment:
Just FYI, no need to change: Could also be done by https://en.cppreference.com/w/cpp/io/basic_ios/exceptions